### PR TITLE
[BugFix] Set _get_pindex_thread_pool as a global thread pool

### PR DESCRIPTION
**Pull request by sevev**
There are two major changes:
1. Set a global thread pool `_get_pindex_thread_pool` to support persisent index get. The main purpose is support parallel get among mutiple L1 files to speed up apply.
2. Add return status check to handle the error code. We don't check the return status of primary index's operation before. Even if some errors happened, we still treat it as normal status, this maybe cause serious problem.

## What type of PR is this:
- [x] BugFix
- [ ] Feature
- [ ] Enhancement
- [ ] Refactor
- [ ] UT
- [ ] Doc
- [ ] Tool

## Checklist:
- [ ] I have added test cases for my bug fix or my new feature
- [ ] This pr will affect users' behaviors
- [ ] This pr needs user documentation (for new or modified features or behaviors)
  - [ ] I have added documentation for my new feature or new function

## Bugfix cherry-pick branch check:
- [x] I have checked the version labels which the pr will be auto-backported to the target branch
  - [x] 3.0
  - [x] 2.5
  - [ ] 2.4
  - [ ] 2.3

### CODE REVIEW

File: be/src/storage/persistent_index.cpp [hunk 1]
```diff
@@ -2888,6 +2888,7 @@ Status PersistentIndex::on_commited() {
     if (_get_thread_pool != nullptr) {
         _get_thread_pool->shutdown();
     }
+    _get_thread_pool.reset();
```
Comment (luohaha) [inline be/src/storage/persistent_index.cpp:2891]: Can we use a global thread pool for this, instead of create and free thread pool every time.

**Review by chaoyli — DISMISSED** [review]

**Comment by sevev** [comment]
@mergify rebase

**Review by trueeyu — DISMISSED** [review]

**Review by trueeyu — DISMISSED** [review]

**Review by decster — APPROVED** [review]

**Review by trueeyu — APPROVED** [review]

**Comment by wanpengfei-git** [comment]
@Mergifyio backport branch-2.5

**Comment by sevev** [comment]
https://github.com/StarRocks/starrocks/pull/25015 backport branch-2.5

**Comment by sevev** [comment]
https://github.com/StarRocks/starrocks/pull/25013 backport branch-3.0

**Comment by sevev** [comment]
@mergify backport branch-3.1

**Comment by sevev** [comment]
branch-3.1, https://github.com/StarRocks/starrocks/pull/25540

**Pull request merged**